### ISSUE TITLE
Properly check for LV2 UI extension URI

### ISSUE THREAD

**Issue by ventosus** [sadko4u/lsp-plugins#32]
Please add a proper check for the LV2 UI extension URI before returning anything potentially *invalid*.

366     const void* lv2ui_extension_data(const char* uri)                                                                               
367     {                                                                                                                    
368         if (!strcmp(uri, LV2_UI__idleInterface))                                                                         
369         {                                                                                                                
370           return &idle_iface;                                                                                            
371         }                                                                                                                
372                                                                                                                          
373         return NULL;                                                                                                     
374     }

**Comment by ventosus** [comment]
And properly register the extension in your *.ttl [1]

    lv2:extensionsData ui:idleInterface ;

[1] http://lv2plug.in/ns/extensions/ui/ui.html#idleInterface

**Comment by ventosus** [comment]
There are at least 3 LV2 UI extensions

* http://lv2plug.in/ns/extensions/ui/ui.html#idleInterface
* http://lv2plug.in/ns/extensions/ui/ui.html#resize
* http://lv2plug.in/ns/extensions/ui/ui.html#showInterface

Hosts that support all of them, are not able to figure out what you support or not.

**Comment by sadko4u** [comment]
Confirmed. Will fix the next day.

**Comment by sadko4u** [comment]
This commit should fix the problem: 83ca69b7f709d0f3356d812a0069c751085f14d4

**Comment by sadko4u** [comment]
Now metadata of the UI looks like:
```
lsp_ui:trigger_midi_mono
        a ui:X11UI ;
        lv2:minorVersion 0 ;
        lv2:microVersion 0 ;
        lv2:requiredFeature urid:map, ui:idleInterface ;
        lv2:optionalFeature ui:parent, ui:resize, ui:noUserResize, lv2ext:instance-access ;
        lv2:extensionData ui:idleInterface ;
        ui:binary <lsp-plugins-lv2.so> ;
        ...
```

**Comment by sadko4u** [comment]
Merged changes into **devel** branch: 595ad1e18430a45c413ecb6652588e71edd73cbd

**Comment by ventosus** [comment]
thanks, looks good

**Comment by sadko4u** [comment]
All changes are available in the 1.1.7 release that was just published.
